### PR TITLE
Cristibm/lab 10 update

### PR DESCRIPTION
**Pull request by CristiBM**
Replace the inline assembly tasks with 3 others that cover registers not being preserved by the callee and the mechanism of setting and restoring the state of the stack frame

### CODE REVIEW

**Review by darius-m — CHANGES_REQUESTED** [review]
The line indentations should be done using tabs, instead of spaces. You could use `sed` to update the lines with a rule like `s/^    /\t/`.

File: laborator/content/interactiune-c-assembly/README.md [hunk 1]
```diff
@@ -197,6 +138,20 @@ La afișare se va afișa și poziția din vector pe care se găsește maximul.
...
 
+### 6. Tutorial: Conservare registre
+În subdirectorul `6-7-regs-preserve/` din arhiva de sarcini a laboratorului găsiți funcția `print_reverse_array()` implementată printr-un simplu loop ce face apeluri repetate ale funcției `printf()`.
+
+Urmăriți codul din fișierul `main.asm`, compilați și rulați programul. Ce s-a întâmplat? Programul rulează la infinit. Acest lucru se întămplă deoarece funcția `printf()` nu conservă valoarea din registrul `ECX`, folosit aici ca și contor.
+
+Decomentați liniile marcate cu `TODO1` și rerulați programul.
+
+### 7. Depanare SEGFAULT
+Decomentați liniile marcate cu `TODO2` în fișierul assembly de la exercițiul anterior. Snippetul de cod realizează un apel al funcției `double_array()`, implementată în C, chiar înainte de afișarea vectorului folosind funcția văzută anterior.
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/README.md:149]:
```suggestion
Decomentați liniile marcate cu `TODO2` în fișierul assembly de la exercițiul anterior. Secvența de cod realizează un apel al funcției `double_array()`, implementată în C, chiar înainte de afișarea vectorului folosind funcția văzută anterior.
```

File: laborator/content/interactiune-c-assembly/README.md [hunk 2]
```diff
@@ -197,6 +138,20 @@ La afișare se va afișa și poziția din vector pe care se găsește maximul.
...
+
+Compilați și rulați programul. Pentru depanarea segfault-ului puteți folosi utilitarul `objdump` pentru a urmări codul assembly corespunzător funcției `double_array()`. Observați care din registrele folosite înainte și după apel sunt modificate de această funcție.
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/README.md:151]:
```suggestion
Compilați și rulați programul. Pentru depanarea segfault-ului puteți folosi utilitarul `objdump` pentru a urmări codul în limbaj de asamblare corespunzător funcției `double_array()`. Observați care din registrele folosite înainte și după apel sunt modificate de această funcție.
```

File: laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+print_reverse_array:
+    push ebp
+    mov ebp, esp
+
+    ; save ebx in callee
+    push ebx
+
+    ; [ebp+8] is array pointer
+    ; [ebp+12] is array length
+
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm:24]:
```suggestion
```

File: laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    push dword [ebx+ecx*4-4]
+    push format_string
+    call printf
+    add esp, 8
+;   pop ecx
+
+    loop next
+
+
+    push newline
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm:42]:
```suggestion
    add esp, 8
;   pop ecx
    loop next

    push newline
```

File: laborator/content/interactiune-c-assembly/3-stack-frame/print-hello.asm [hunk 5]
```diff
@@ -0,0 +1,19 @@
+extern printf
+
+section .data
+    message db "Hello", 0
+
+section .text
+
+global print_hello
+
+# TODO: Adăugați instrucțiunea lipsă
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/3-stack-frame/print-hello.asm:10]:
```suggestion
;   TODO: Adăugați instrucțiunea lipsă
```

File: laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    ; [ebp+8] is array pointer
+    ; [ebp+12] is array length
+
+    mov ebx, [ebp+8]
+    mov ecx, [ebp+12]
+    xor eax, eax
+
+next:
+
+; TODO1: Decomentați următoarele două linii comentate
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm:31]:
```suggestion
;   TODO1: Decomentați următoarele două linii comentate
```

File: laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+global main
+
+main:
+    push ebp
+    mov ebp, esp
+
+    mov edx, [myarray_len]
+    lea eax, [myarray]
+
+; TODO2: Decomentați acest snippet de cod
```
Comment (darius-m) [inline laborator/content/interactiune-c-assembly/6-7-regs-preserve/main.asm:61]:
```suggestion
;   TODO2: Decomentați acest snippet de cod
```

**Review by darius-m — CHANGES_REQUESTED** [review]
Mostly looks good. I've added some small styling suggestions. If you decide to accept them, please use the batch add option, otherwise each of them will generate a message for the PR.

Also, please address the coding style errors / warnings in the automatic checks (ignoring the licensing messages).

**Comment by CristiBM** [comment]
> Mostly looks good. I've added some small styling suggestions. If you decide to accept them, please use the batch add option, otherwise each of them will generate a message for the PR.
> 
> Also, please address the coding style errors / warnings in the automatic checks (ignoring the licensing messages).

Solved the comments and coding style errors with the exception of the licensing messages and one regarding an allegedly 'misspelled' word in a code comment.

**Review by darius-m — APPROVED** [review]
Looks good now

**Pull request merged**